relayer/chains/cosmos/event_parser_test.go

```diff
@@ -0,0 +1,315 @@
+package cosmos
+
+import (
+	"encoding/hex"
+	"strings"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	clienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+	conntypes "github.com/cosmos/ibc-go/v3/modules/core/03-connection/types"
+	chantypes "github.com/cosmos/ibc-go/v3/modules/core/04-channel/types"
+	"github.com/cosmos/relayer/v2/relayer/processor"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+)
+
+func TestParsePacket(t *testing.T) {
+	testPacketTimeoutHeight := "1-1245"
+	testPacketTimeoutTimestamp := "1654033235600000000"
+	testPacketSequence := "1"
+	testPacketDataHex := "0123456789ABCDEF"
+	testPacketSrcChannel := "channel-0"
+	testPacketSrcPort := "port-0"
+	testPacketDstChannel := "channel-1"
+	testPacketDstPort := "port-1"
+
+	packetEventAttributes := []sdk.Attribute{
+		{
+			Key:   chantypes.AttributeKeySequence,
+			Value: testPacketSequence,
+		},
+		{
+			Key:   chantypes.AttributeKeyDataHex,
+			Value: testPacketDataHex,
+		},
+		{
+			Key:   chantypes.AttributeKeyTimeoutHeight,
+			Value: testPacketTimeoutHeight,
+		},
+		{
+			Key:   chantypes.AttributeKeyTimeoutTimestamp,
+			Value: testPacketTimeoutTimestamp,
+		},
+		{
+			Key:   chantypes.AttributeKeySrcChannel,
+			Value: testPacketSrcChannel,
+		},
+		{
+			Key:   chantypes.AttributeKeySrcPort,
+			Value: testPacketSrcPort,
+		},
+		{
+			Key:   chantypes.AttributeKeyDstChannel,
+			Value: testPacketDstChannel,
+		},
+		{
+			Key:   chantypes.AttributeKeyDstPort,
+			Value: testPacketDstPort,
+		},
+	}
+
+	parsed := new(packetInfo)
+	parsed = parsed.parsePacketInfo(zap.NewNop(), "", packetEventAttributes)
+
+	require.Equal(t, uint64(1), parsed.packet.Sequence, "packet sequence does not match")
+	require.Equal(t, strings.ToUpper(testPacketDataHex), strings.ToUpper(hex.EncodeToString(parsed.packet.Data)), "packet data does not match")
+
+	require.Equal(t, uint64(1), parsed.packet.TimeoutHeight.RevisionNumber, "packet timeout height revision number does not match")
+	require.Equal(t, uint64(1245), parsed.packet.TimeoutHeight.RevisionHeight, "packet timeout height revision height does not match")
+
+	require.Equal(t, uint64(1654033235600000000), parsed.packet.TimeoutTimestamp, "packet timeout timestamp does not match")
+
+	require.Equal(t, testPacketSrcChannel, parsed.packet.SourceChannel, "packet source channel does not match")
+	require.Equal(t, testPacketSrcPort, parsed.packet.SourcePort, "packet source port does not match")
+	require.Equal(t, testPacketDstChannel, parsed.packet.DestinationChannel, "packet destination channel does not match")
+	require.Equal(t, testPacketDstPort, parsed.packet.DestinationPort, "packet destination port does not match")
```

Comment:
Once a struct gets this many fields, I often reach to [go-cmp](https://pkg.go.dev/github.com/google/go-cmp/cmp) to simplify comparison tests like this:

```go
parsed := ...

require.Empty(t, cmp.Diff(parsed, packetInfo{
  packet: packet{
    Sequence: 1,
    // ...
  },
})
```

It is easy to explicitly configure cmp.Diff to ignore one particular field, or to otherwise set custom comparison rules. 

But if there is a good reason to keep it the way it is, that's fine.

Reply:
looks great. done